File: leetcode/cpp/566_Reshape_the_Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(mat), len(mat[0])
        if m * n != r * c: return mat

        mat2 = [[0] * c for _ in range(r)]

        x = 0
        for i in range(m):
            for j in range(n):
                mat2[x // c][x % c] = mat[i][j]
                x += 1

        return mat2

File: leetcode/cpp/566_Reshape_the_Matrix.cpp
#include <vector>
using namespace std;
class Solution {
public:
	vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
		vector<vector<int>> mat2(r, vector<int> (c, 0));	

		size_t m = mat.size(), n = mat[0].size();
		if (r * c != m * n) return mat;

		int x = 0;
		for (int i = 0; i < m; ++i) {
			for (int j = 0; j < n; ++j) {
				mat2[x / c][x % c] = mat[i][j];
				++x;
			}
		}

		return mat2;
	}
};
